src/models/fields/base-field-configuration.ts

```diff
@@ -1,38 +1,241 @@
+      is: 'integer',
+      then: Joi.number().integer().default(0),
+    })
+    .when(Joi.ref('type'), {
+      is: 'boolean',
+      then: Joi.boolean().default(false),
+    }),
   valueOptions: Joi.array().items(Joi.object().keys({
-    text: Joi.string().required(),
+    text: stringOrTranslationsSchema.required(),
     value: baseFieldSchema.required().description('Will be passed through parser'),
-  })).allow(null)
-    .description('If null, inputType should not be select or radio. If not null, input type should be select or radio'),
-  prefix: Joi.string().description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
-  suffix: Joi.string().description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
+  })).allow(null).default(null)
+    .description('If null, inputType should not be select or radio. If not null, input type should be select or radio. Default: null'),
+  regex: Joi.string().description('If provided, type should be string and the provided value should adhere to this regex'),
+  lowerbound: Joi.number().description('If provided, type should be number or integer and provided value should not be lower than this value'),
+  upperbound: Joi.number().description('If provided, type should be number or integer and provided value should not be higher than this value'),
+  showIf: Joi.object().keys({
+    key: Joi.string().required(),
+    value: Joi.alternatives(
+      Joi.number(),
+      Joi.string(),
+      Joi.boolean(),
+    ).required(),
+  }).description('Show this field if other field with provided key is set to provided value. Referenced field must exists already'),
+  prefix: stringOrTranslationsSchema.description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
+  suffix: stringOrTranslationsSchema.description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
   hint: Joi.string().allow('').description('As shown near the input field'),
 })
   .tag('baseFieldConfiguration')
   .description('Defines which data can be stored in form fields.');
 
-interface ValueOption {
-  text: string;
-  value: BaseField;
+interface ValueOption<T extends BaseField = BaseField> {
+  // string for backwards compatibility, shown at all locales
+  text: StringOrTranslations;
+  value: T;
 }
 
-interface BaseFieldConfiguration {
+interface SharedBaseFieldConfiguration {
+  /**
+   * key used to store the field value
+   * @pattern ^[a-z][a-zA-Z\d]*$
+   */
   key: string;
-  name: string;
-  inputType?: 'text' | 'textarea' | 'select' | 'radio' | 'switch' | 'checkbox' | 'file' | 'files';
-  valueOptions?: ValueOption[] | null;
-  prefix?: string;
-  suffix?: string;
-  hint?: string;
+  /**
+   * name of the field to be shown in UI
+   */
+  name: StringOrTranslations;
+  /**
+   * show this field in UI only if value of another form field, identified by provided `key`, is
```

Comment:
> of another form field

I propose to make that 'of an earlier form field', to rule out circular references. I know it's also possible to do more advanced analysis and allow later referencing form fields, but I also think it does not make sense in the UI that an earlier component is shown when you change something further down.